src/main/java/com/autosale/shop/security/JwtTokenUtil.java

```diff
@@ -0,0 +1,91 @@
+package com.autosale.shop.security;
+
+import com.autosale.shop.model.User;
+import com.autosale.shop.service.UserService;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.JwtException;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.security.Keys;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+@Component
+@RequiredArgsConstructor
+public class JwtTokenUtil {
+
+    @Value("${jwt.secret}")
+    private String SECRET_KEY;
+
+    private final UserService userService;
+
+
+    public String generateAccessToken(User user) {
+        return Jwts.builder().claim("id", user.getId().toString())
+                .claim("username", user.getUserName())
+                .claim("role", user.getRole())
+                .setId(UUID.randomUUID().toString())
+                .setSubject(String.format("%s,%s", user.getId(), user.getUserName()))
```

Comment:
do we need this susbject? You provide the same info through claims

Reply:
Refactored